src/main/java/com/acme/edu/Logger.java

```diff
@@ -2,111 +2,281 @@
+            processAllBuffers();
+        }
+
+        lastLoggedType = LastLoggedType.BYTE;
     }
 
-    private static  void checkIntOverFlow(int message) {
+    public static void log(char message) {
+        processAllBuffers();
+        save(formatMessage(CHAR_PREFIX, message));
+        lastLoggedType = LastLoggedType.CHAR;
+    }
+
+    public static void log(boolean message) {
+        processAllBuffers();
+        save(formatMessage(PRIMITIVE_PREFIX, message));
+        lastLoggedType = LastLoggedType.BOOL;
+    }
+
+    public static void log(String message) {
+        if (!lastLoggedType.equals(LastLoggedType.STRING) || !message.equals(lastString)) {
+            processAllBuffers();
+        }
+
+        lastString = message;
+        ++repeatableStringCounter;
+
+        lastLoggedType = LastLoggedType.STRING;
+    }
+
+    public static void log(Object message) {
+        processAllBuffers();
+        save(formatMessage(OBJECT_PREFIX, message));
+        lastLoggedType = LastLoggedType.OBJ;
+    }
+    //endregion
+
+    //region int methods
+    private static void checkIntOverFlow(int message) {
+        if (message == Integer.MAX_VALUE) {
+            increaseIntegerThresholdValueCounter();
+            return;
+        }
+
+        if (message == Integer.MIN_VALUE) {
+            decreaseIntegerThresholdValueCounter();
+            return;
+        }
+
         if (message > 0) {
             checkMaxIntOverFlow(message);
         } else {
             checkMinIntOverFlow(message);
         }
     }
 
-    private static void checkMaxIntOverFlow(int message /*>=0*/) {
-        long diff = Integer.MAX_VALUE-(long)intBuffer;
-        if ( diff > message) {
+    private static void checkMaxIntOverFlow(int message) {
+        long diff = Integer.MAX_VALUE - (long) intBuffer;
+        if (diff > message) {
             intBuffer += message;
-        } else { // only if intBuffer > 0
-            intBuffer = (int)(message-diff);
-            if (intMaxMinValueCounter > 0) {
-                --intBuffer;
-            }
-            ++intMaxMinValueCounter;
+        } else {
+            intBuffer = (int) (message - diff);
+            increaseIntegerThresholdValueCounter();
         }
     }
 
     private static void checkMinIntOverFlow(int message) {
-        long diff = (long)intBuffer-Integer.MIN_VALUE;
-        if ( diff > Math.abs(message)) {
+        long diff = (long) intBuffer - Integer.MIN_VALUE;
+        if (diff > Math.abs((long) message)) {
             intBuffer += message;
-        } else { // only if intBuffer < 0
-            intBuffer = (int)(message+diff);
-            if (intMaxMinValueCounter < 0) {
-                --intBuffer;
+        } else {
+            intBuffer = (int) (message + diff);
+            decreaseIntegerThresholdValueCounter();
+        }
+    }
+
+    private static void increaseIntegerThresholdValueCounter() {
+        if (integerThresholdValueCounter < 0) {
+            --intBuffer;
```

Comment:
Isn't it possible for intBuffer to take MIN_VALUE here, not only in case covered by decreaseIntegerThresholdValueCounter()? For example,
integerThresholdValueCounter = -1
intBuffer = MIN_VALUE+1
message = MAX_VALUE